test/performance/io/format_fastq_benchmark.cpp

```diff
@@ -0,0 +1,285 @@
+
+    // benchmark
+    std::string id{};
+    std::vector<dna5> seq{};
+    std::vector<phred42> qual{};
+
+    for (auto _ : state)
+    {
+        sequence_file_input fin{file_name};
+
+        // read all records and store in internal buffer
+        auto it = fin.begin();
+        while (it != fin.end())
+            ++it;
+    }
+
+    // remove temporary file
+    const int remove_success = std::remove(file_name.c_str());
+    if (remove_success != 0) // sic
+    {
+        std::perror("Error removing temporary file \"tmp.fastq\" for fastq_read_from_disk.");
+        std::cout << "aborting..." << std::endl;
+        exit(1);
+    }
+}
+
+// ============================================================================
+// seqan2 comparison
+// ============================================================================
+#if SEQAN2_HAS_SEQAN3
+
+template<std::size_t n_entries_in_file = N_ENTRIES_IN_FILE>
+void fastq_read_from_disk_seqan2(benchmark::State & state)
+{
+    using namespace seqan;
+
+    // create
+    std::filesystem::path tmp_dir = std::filesystem::temp_directory_path();
+    std::string const file_name = tmp_dir/"tmp.fastq";
+
+    std::ofstream ostream{file_name};
+    bool open_success = ostream.is_open();
+
+    if (!open_success)
+    {
+        std::perror("Error creating temporary file \"tmp.fastq\" for fastq_read_from_disk_seqan2.");
+        std::cout << "aborting..." << std::endl;
+        exit(1);
+    }
+
+    ostream << get_file(n_entries_in_file);
+    ostream.close();
+
+    // benchmark
+    StringSet<String<char>> ids{};
+    StringSet<String<Dna5>> seqs{};
+    StringSet<String<char>> quals{};
+
+    for (auto _ : state)
+    {
+        SeqFileIn seqFileIn(file_name.c_str());
+        readRecords(ids, seqs, quals, seqFileIn);
+
+        clear(ids);
+        clear(seqs);
+        clear(quals);
+    }
+
+    // delete
+    const int remove_success = std::remove(file_name.c_str());
+    if (remove_success != 0)
+    {
+        std::perror("Error removing temporary file \"tmp.fastq\" for fastq_read_from_disk_seqan2.");
+        std::cout << "aborting..." << std::endl;
+        exit(1);
+    }
+}
+
+#endif
+
+BENCHMARK(fastq_write);
+
+#if SEQAN2_HAS_SEQAN3
+
+    BENCHMARK_TEMPLATE(fastq_read_from_stream, 1000);
+    BENCHMARK_TEMPLATE(fastq_read_from_stream_seqan2, 1000);
+    BENCHMARK_TEMPLATE(fastq_read_from_disk, 1000);
+    BENCHMARK_TEMPLATE(fastq_read_from_disk_seqan2, 1000);
+
+    BENCHMARK_TEMPLATE(fastq_read_from_stream, 5000);
+    BENCHMARK_TEMPLATE(fastq_read_from_stream_seqan2, 5000);
+    BENCHMARK_TEMPLATE(fastq_read_from_disk, 5000);
+    BENCHMARK_TEMPLATE(fastq_read_from_disk_seqan2, 5000);
+
+    BENCHMARK_TEMPLATE(fastq_read_from_stream, 10000);
+    BENCHMARK_TEMPLATE(fastq_read_from_stream_seqan2, 10000);
+    BENCHMARK_TEMPLATE(fastq_read_from_disk, 10000);
+    BENCHMARK_TEMPLATE(fastq_read_from_disk_seqan2, 10000);
+
+#else
+
+    BENCHMARK_TEMPLATE(fastq_read_from_stream);
```

Comment:
Travis performance test:

```
/home/travis/build/seqan/seqan3/test/performance/io/format_fastq_benchmark.cpp:280:46: error: ISO C++11 requires at least one argument for the "..." in a variadic macro [-Werror]
     BENCHMARK_TEMPLATE(fastq_read_from_stream);
                                              ^
```

Can you extract all `seqan3`-only  performance tests and move them out of the `ifdef`

Reply:
fixed